user-operator/src/test/java/io/strimzi/operator/user/operator/KafkaUserQuotasIT.java

```diff
@@ -170,27 +176,166 @@ public void testDeletion()  {
         assertThat(updated.getJsonObject("config").getString("producer_byte_rate"), is(notNullValue()));
+        assertThat(updated.getJsonObject("config").getString("request_percentage"), is(notNullValue()));
         assertThat(updated.getJsonObject("config").getString("producer_byte_rate"), is("4000"));
-        assertThat(updated.getJsonObject("config").getString("consumer_byte_rate"), is("3000"));
+        assertThat(updated.getJsonObject("config").getString("request_percentage"), is("40"));
 
-        original = new JsonObject().put("version", 1).put("config", new JsonObject());
-        quotas.setProducerByteRate(null);
-        quotas.setConsumerByteRate(1000);
-        updated = new JsonObject(new String(kuq.updateUserJson(original.encode().getBytes(StandardCharsets.UTF_8), quotas), StandardCharsets.UTF_8));
+        KafkaUserQuotas quotas5 = new KafkaUserQuotas();
+        quotas5.setConsumerByteRate(20000);
+        quotas5.setProducerByteRate(40000);
+        quotas5.setRequestPercentage(50);
+
+        updated = new JsonObject(new String(kuq.createOrUpdateUserJson(createdBytes, quotas5), StandardCharsets.UTF_8));
         assertThat(updated.getJsonObject("config").getString("consumer_byte_rate"), is(notNullValue()));
+        assertThat(updated.getJsonObject("config").getString("producer_byte_rate"), is(notNullValue()));
+        assertThat(updated.getJsonObject("config").getString("request_percentage"), is(notNullValue()));
+        assertThat(updated.getJsonObject("config").getString("consumer_byte_rate"), is("20000"));
+        assertThat(updated.getJsonObject("config").getString("producer_byte_rate"), is("40000"));
+        assertThat(updated.getJsonObject("config").getString("request_percentage"), is("50"));
+    }
+
+    @Test
+    public void testReconcileCreate(VertxTestContext testContext)  {
+        String user = "createTestUser";
+        KafkaUserQuotas quotas = new KafkaUserQuotas();
+        quotas.setConsumerByteRate(2_000_000);
+        quotas.setProducerByteRate(1_000_000);
+        quotas.setRequestPercentage(50);
+
+        testContext.verify(() -> assertThat(kuq.exists(user), is(false)));
+
+        Checkpoint async = testContext.checkpoint();
+        kuq.reconcile(user, quotas)
+                .setHandler(testContext.succeeding(res -> {
+                    testContext.verify(() -> assertThat(kuq.exists(user), is(true)));
+                    testContext.verify(() -> assertThat(kuq.getQuotas(user).getJsonObject("config").getString("consumer_byte_rate"), is("2000000")));
+                    testContext.verify(() -> assertThat(kuq.getQuotas(user).getJsonObject("config").getString("producer_byte_rate"), is("1000000")));
+                    testContext.verify(() -> assertThat(kuq.getQuotas(user).getJsonObject("config").getString("request_percentage"), is("50")));
+                    testContext.verify(() -> assertThat(kuq.isPathExist("/config/users/" + user), is(true)));
```

Comment:
maybe we can do just one `verify` with all `assertThat` inside as done in #2651 ?

Reply:
Better?

Reply:
yep :-)